buildpacks/ruby-bundler/bin/detect

```diff
@@ -1,16 +1,35 @@
 #!/usr/bin/env bash
 set -eo pipefail
 
+# 1. CHECK IF APPLICABLE 
 if [[ ! -f Gemfile ]]; then
    exit 100
 fi
 
-plan=$2
-version=2.5.1
+# 2. GET ARGS
+plan_path=$2
 
+# 3. GET CONFIGURATION
+version=2.5.1
 if [[ -f .ruby-version ]]; then
     version=$(cat .ruby-version | tr -d '[:space:]')
 fi
 
-echo "provides = [{ name = \"ruby\" }]" > "$plan"
-echo "requires = [{ name = \"ruby\", version = \"$version\" }]" >> "$plan"
\ No newline at end of file
+# 4. DECLARE DEPENDENCIES
+cat >> "${plan_path}" <<EOL
+# Buildpack provides this dependency
+#
+# NOTE: The dependency is provided during the 'build' process.
+#
+[[provides]]
+name = "ruby"
+
+# Buildpack requires this dependency
+#
+# NOTE: Everything aside from the name is simply additional information the buildpack that provides this dependency
```

Comment:
Are you saying that the other fields aside from `name` can be anything? Metadata used for resolving the dependency.

Slight rewording perhaps (here and in the other files that have this note):
> Everything aside from `name` is simply additional metadata that the providing buildpack can use to resolve the dependency

Reply:
Correct. I used your wording but replaced `metadata` with `information` because there is an actual `metadata` field. Information IMO encompasses any other field, including `version`.